Alerting: Provisioning Status Differentiation for ALL resources

rodrigopk opened the pull request:
**What is this feature?**
Refactors the provisioning badge component in order to differentiate resources provisioned from Prometheus or Mimir from other sources. All resources (contact points, templates, policies) will display the appropriate badge with explanatory tooltips.
Also added changes to ensure that the `provenance` value is passed down correctly in the points where the `ProvisioningBadge` component is used. 


https://github.com/user-attachments/assets/f6ca5fac-5df5-423b-bb18-1b1e8a4d6d65



**How to test these changes**
- ❗You will need to import a mimir config to see imported contact points and templates, reach out via slack for a script to facilitate that
- Navigate to Alerting -> Contact points
- Confirm that imported contact points and templates show the new badge
- Confirm that other provisioned contact points show the regular "Provisioned" badge
- Confirm that non provisioned contact points do not show the badge
- Confirm that provisioned contact points are not editable
- Confirm the same for notification templates

**Why do we need this feature?**

As part of the work for the phase 2 of the Single Alertmanager, we need to better distinguish from traditional (file-based) provisioning

**Who is this feature for?**

Users who use provisioned resources coming from Prometheus or Mimir 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/alerting-squad/issues/1341

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

rodrigopk committed:
Show different badge for converted prometheus provisioned resource

rodrigopk committed:
Update contact points and templates to populate provenance

rodrigopk committed:
Update notification policies

rodrigopk committed:
Handle non k8s contact point in ContactPointHeader

rodrigopk committed:
Fix provenance check in enhanceContactPointsWithMetadata

rodrigopk committed:
Update translations

rodrigopk committed:
Fix unused import

gillesdemey commented on public/app/features/alerting/unified/types/provenance.ts:
```diff
@@ -0,0 +1,6 @@
+export enum Provenance {
```
nit: perhaps `KnownProvenance` or something to indicate this isn't a fixed list? Users can choose their own string if they want.

rodrigopk committed:
Refactor provenance enum

rodrigopk committed:
Derive provisioned status from provenance in Route type

rodrigopk committed:
Remove unused imports

rodrigopk committed:
Treat PROVENANCE_NONE as no provenance in isRouteProvisione

rodrigopk committed:
Rename KnownProvenance.None to .Empty to avoid confusion

rodrigopk committed:
Change copy text for resources with converted_prometheus provenance

rodrigopk committed:
Derive provisioned status from provenance in GrafanaManagedContactPoint

rodrigopk committed:
Fix linter errors

rodrigopk committed:
Extract helper method to check if contact point is provisioned

gillesdemey reviewed (approved):
LGTM, just a small nit – awesome work Rodrigo! 🚀

rodrigopk committed:
Replace string literal with constant

konrad147 reviewed (approved):
LGTM: one non-blocking nit

soniaAguilarPeiron commented on public/app/features/alerting/unified/components/notification-policies/useNotificationPolicyRoute.ts:
```diff
@@ -236,6 +238,7 @@ function k8sRoutesToRoutes(routes: ComGithubGrafanaGrafanaPkgApisAlertingNotific
         resourceVersion: route.metadata.resourceVersion,
         name: route.metadata.name,
       },
+      provenance: getAnnotation(route, K8sAnnotations.Provenance) || Provenance.None,
```
ok, see discussion [here](https://raintank-corp.slack.com/archives/C08PSLT263E/p1768234409451909):
### Summary: 
We only have 4 types of provenance:` none, file, api, prometheus_convert`, where none represents empty sting in the database and means that this resource is not provisioned.

rodrigopk replied:
@soniaAguilarPeiron refactored in https://github.com/grafana/grafana/pull/115773/changes/b718c41a46ad300bd8ca76527f28d96ecca21a14

rodrigopk committed:
Refactor KnownProvenance enum values

Refactored the KnownProvenance enum to better reflect the known provenances defined by the backend.
Also refactored the methods where we assert if a resource is provisioned to better reflect the cases for which a provenance value reflects no provisioning.
A resource is considered not provisioned when the provenance is equal to '', 'none' or undefined.

rodrigopk committed:
Use provenance to infer provenance status for Templates

Refactored useNotificationTemplateMetadata to use only provenance value, and extracted method used to assert if resource is provisioned or not to k8s/utils in order to be more resource agnostic.

soniaAguilarPeiron commented on public/app/features/alerting/unified/types/knownProvenance.ts:
```diff
@@ -1,6 +1,5 @@
 export enum KnownProvenance {
-  None = 'none',
-  Empty = '',
+  None = '',
```
In the frontend we only receive 'none', no empty strings are returned by the BE, [empty strings are translated to 'none'](https://github.com/grafana/grafana/blob/896a56ada8ea7a03090591848e269bdc1f42808d/apps/alerting/notifications/pkg/apis/alertingnotifications/v0alpha1/receiver_ext.go#L21-L29)

rodrigopk replied:
@soniaAguilarPeiron Refactored in https://github.com/grafana/grafana/commit/00fc62ccf5fecfa3b781ab4163d016d29f190b2f 👍

rodrigopk committed:
Replace empty string with 'none' for KnownProvenance enum

The empty string valye for provenance gets mapped to the string literal 'none' before being passed down in the api response, therefore we can use only 'none'

rodrigopk committed:
Replace PROVENANCE_NONE with KnownProvenance.None

Replaced the constant PROVENANCE_NONE with the KnownProvenance.None enum value since the values where duplicated

rodrigopk committed:
Fix JSDoc

rodrigopk committed:
Change copy text for ProvisioningBadge

rodrigopk committed:
Add missing tooltip in notification policy badge

rodrigopk committed:
Add missing tooltip in TemplatesTable badge

soniaAguilarPeiron reviewed (approved):
LGTM! :shipit: 
Great work Rodrigo!!! 🚀  👏

konrad147 reviewed (approved):
LGTM!

soniaAguilarPeiron committed:
Merge branch 'main' into rodrigopk/provisioning-status-differentiation-for-all-resources

soniaAguilarPeiron committed:
fix conflicts
